src/PyF/Internal/Meta.hs

```diff
@@ -0,0 +1,229 @@
+toExp d (Expr.NegApp _ e _)                  = TH.AppE (TH.VarE 'negate) (toExp d . unLoc $ e)
+-- toExp (Expr.Lambda _ ps e)                    = TH.LamE (fmap toPat ps) (toExp e)
+-- toExp (Expr.Let _ bs e)                       = TH.LetE (toDecs bs) (toExp e)
+-- toExp (Expr.If _ a b c)                       = TH.CondE (toExp a) (toExp b) (toExp c)
+-- toExp (Expr.MultiIf _ ifs)                    = TH.MultiIfE (map toGuard ifs)
+-- toExp (Expr.Case _ e alts)                    = TH.CaseE (toExp e) (map toMatch alts)
+-- toExp (Expr.Do _ ss)                          = TH.DoE (map toStmt ss)
+-- toExp e@Expr.MDo{}                            = noTH "toExp" e
+-- toExp (Expr.Tuple _ Exts.Boxed xs)            = TH.TupE (fmap toTupEl xs)
+-- toExp (Expr.Tuple _ Exts.Unboxed xs)          = TH.UnboxedTupE (fmap toTupEl xs)
+-- toExp e@Expr.TupleSection{}                   = noTH "toExp" e
+-- toExp (Expr.List _ xs)                        = TH.ListE (fmap toExp xs)
+toExp d (Expr.HsPar _ e)                           = TH.ParensE (toExp d . unLoc $ e)
+-- toExp (Expr.LeftSection _ e o)                = TH.InfixE (Just . toExp $ e) (toExp o) Nothing
+-- toExp (Expr.RightSection _ o f)               = TH.InfixE Nothing (toExp o) (Just . toExp $ f)
+toExp _ (Expr.RecordCon _ name HsRecFields{rec_flds})
+  = TH.RecConE (toName . unLoc $ name) (fmap toFieldExp rec_flds)
+-- toExp (Expr.RecUpdate _ e xs)                 = TH.RecUpdE (toExp e) (fmap toFieldExp xs)
+-- toExp (Expr.EnumFrom _ e)                     = TH.ArithSeqE $ TH.FromR (toExp e)
+-- toExp (Expr.EnumFromTo _ e f)                 = TH.ArithSeqE $ TH.FromToR (toExp e) (toExp f)
+-- toExp (Expr.EnumFromThen _ e f)               = TH.ArithSeqE $ TH.FromThenR (toExp e) (toExp f)
+-- toExp (Expr.EnumFromThenTo _ e f g)           = TH.ArithSeqE $ TH.FromThenToR (toExp e) (toExp f) (toExp g)
+-- toExp (Expr.ListComp _ e ss)                  = TH.CompE $ map convert ss ++ [TH.NoBindS (toExp e)]
+--  where
+--   convert (Expr.QualStmt _ st)                = toStmt st
+--   convert s                                   = noTH "toExp ListComp" s
+-- toExp (Expr.ExpTypeSig _ e t)                 = TH.SigE (toExp e) (toType t)
+toExp dynFlags e = todo "toExp" (showSDocDebug dynFlags . ppr $ e)
+
+todo :: (Show e) => String -> e -> a
+todo fun thing = error . concat $ [moduleName, ".", fun, ": not implemented: ", show thing]
+
+noTH :: (Show e) => String -> e -> a
+noTH fun thing = error . concat $ [moduleName, ".", fun, ": no TemplateHaskell for: ", show thing]
+
+moduleName :: String
+moduleName = "PyF.Internal.Meta"
+
+baseDynFlags :: [GhcTH.Extension] -> DynFlags
+baseDynFlags exts =
+  let enable = GhcTH.TemplateHaskellQuotes : exts
+  in foldl xopt_set (defaultDynFlags fakeSettings fakeLlvmConfig) enable
+
+translateTHtoGHCExt :: TH.Extension -> GhcTH.Extension
+translateTHtoGHCExt TH.Cpp                               = GhcTH.Cpp
```

Comment:
This stuff will break on each new GHC version because of added extension.

I still don't know how to handle that correctly (I had a wildcard in the previous version)